### PR TITLE
fix(sql): UnsupportedOperationException on join with constant non-boolean filter expression

### LINKED ISSUE #3323: Unexpected Internal Error: UnsupportedOperationException
### Describe the bug

I feel it somehow misses catching the exception when executing:
```
select count(*) from telemetry as a inner join telemetry as b on a.event=b.event where now();
```
```
java.lang.UnsupportedOperationException
	at io.questdb.griffin.engine.functions.TimestampFunction.getBool(TimestampFunction.java:51)
	at io.questdb.griffin.SqlCodeGenerator.generateJoins(SqlCodeGenerator.java:1750)
	at io.questdb.griffin.SqlCodeGenerator.generateSelect(SqlCodeGenerator.java:2676)
	at io.questdb.griffin.SqlCodeGenerator.generateQuery0(SqlCodeGenerator.java:2272)
	at io.questdb.griffin.SqlCodeGenerator.generateQuery(SqlCodeGenerator.java:2259)
	at io.questdb.griffin.SqlCodeGenerator.generateSubQuery(SqlCodeGenerator.java:3620)
	at io.questdb.griffin.SqlCodeGenerator.generateSelectGroupBy(SqlCodeGenerator.java:3114)
	at io.questdb.griffin.SqlCodeGenerator.generateSelect(SqlCodeGenerator.java:2665)
	at io.questdb.griffin.SqlCodeGenerator.generateQuery0(SqlCodeGenerator.java:2272)
	at io.questdb.griffin.SqlCodeGenerator.generateQuery(SqlCodeGenerator.java:2259)
	at io.questdb.griffin.SqlCodeGenerator.generate(SqlCodeGenerator.java:186)
	at io.questdb.griffin.SqlCompiler.generate(SqlCompiler.java:2731)
	at io.questdb.griffin.SqlCompiler.compileUsingModel(SqlCompiler.java:1322)
	at io.questdb.griffin.SqlCompiler.compileInner(SqlCompiler.java:1271)
	at io.questdb.griffin.SqlCompiler.compileBatch(SqlCompiler.java:322)
	at io.questdb.cutlass.pgwire.PGConnectionContext.processQuery(PGConnectionContext.java:2413)
	at io.questdb.cutlass.pgwire.PGConnectionContext.parse(PGConnectionContext.java:1683)
	at io.questdb.cutlass.pgwire.PGConnectionContext.handleClientOperation(PGConnectionContext.java:418)
	at io.questdb.cutlass.pgwire.PGJobContext.handleClientOperation(PGJobContext.java:89)
	at io.questdb.cutlass.pgwire.PGWireServer$1.lambda$$0(PGWireServer.java:91)
	at io.questdb.network.AbstractIODispatcher.processIOQueue(AbstractIODispatcher.java:189)
	at io.questdb.cutlass.pgwire.PGWireServer$1.run(PGWireServer.java:128)
	at io.questdb.mp.Worker.run(Worker.java:118)
]
```

If you remove the `join`
```
select count(*) from telemetry where now();
```

it gives the correct alert: 
```
DatabaseError: boolean expression expected
```

### To reproduce

_No response_

### Expected Behavior

_No response_

### Environment

```markdown
- **QuestDB version**: 7.1.1
- **OS**: ubuntu 20.04
- **Browser**:
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Fixes #3323

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 10 / 10 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|10|10|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 10 / 10 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|10|10|100.00%|

File: core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java [hunk 1]
```diff
@@ -1748,16 +1748,20 @@ private RecordCursorFactory generateJoins(QueryModel model, SqlExecutionContext
...
+                        // this would have been JoinRecordMetadata, which is new instance anyway
+                        // we have to make sure that this metadata is safely transitioned
+                        // to empty cursor factory
+                        JoinRecordMetadata metadata = (JoinRecordMetadata) master.getMetadata();
+                        metadata.incrementRefCount();
+                        RecordCursorFactory factory = new EmptyTableRecordCursorFactory(metadata);
+                        Misc.free(master);
+                        return factory;
+                    }
+                } catch (UnsupportedOperationException e) {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java:1763]: we are not meant to catch these exceptions, there should be a better way to handle this

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java (hunk 1 above)
Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java]: Good idea. Changed to a type check in c445e862d6f67c77500ddc2f64e9f902111060b3

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|4|4|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|4|4|100.00%|

**Pull request merged**